connector/connect/client/jvm/src/test/scala/org/apache/spark/sql/connect/client/util/RemoteSparkSession.scala

```diff
@@ -117,6 +117,7 @@ trait RemoteSparkSession
   override def beforeAll(): Unit = {
     super.beforeAll()
     SparkConnectServerUtils.start()
+    System.setProperty("arrow.memory.debug.allocator", false.toString)
```

Comment:
If `arrow.memory.debug.allocator` is true, 

https://github.com/apache/arrow/blob/798b417ec342051f3ff30b9953dcb1f1b71dc9e8/java/memory/memory-core/src/main/java/org/apache/arrow/memory/BaseAllocator.java#L428-L438

The memory leak may be covered, let's set "arrow.memory.debug.allocator" to false and use GA check it

Reply:
run `build/sbt "connect-client-jvm/testOnly *ClientE2ETestSuite"` with this change:

```
[info] ClientE2ETestSuite:
[info] - test spark result schema (58 milliseconds)
[info] - test spark result array (42 milliseconds)
[info] - simple dataset test (210 milliseconds)
[info] - simple udf test (252 milliseconds)
[info] org.apache.spark.sql.ClientE2ETestSuite *** ABORTED *** (8 seconds, 283 milliseconds)
[info]   "Memory was leaked by query. Memory leaked: (272)
[info]   Allocator(ROOT) 0/272/272/9223372036854775807 (res/actual/peak/limit)
[info]   " contained "Memory was leaked by query. Memory leaked" (RemoteSparkSession.scala:160)
[info]   org.scalatest.exceptions.TestFailedException:
[info]   at org.scalatest.Assertions.newAssertionFailedException(Assertions.scala:472)
[info]   at org.scalatest.Assertions.newAssertionFailedException$(Assertions.scala:471)
[info]   at org.scalatest.Assertions$.newAssertionFailedException(Assertions.scala:1231)
[info]   at org.scalatest.Assertions$AssertionsHelper.macroAssert(Assertions.scala:1295)
[info]   at org.apache.spark.sql.connect.client.util.RemoteSparkSession.afterAll(RemoteSparkSession.scala:160)
[info]   at org.apache.spark.sql.connect.client.util.RemoteSparkSession.afterAll$(RemoteSparkSession.scala:155)
[info]   at org.apache.spark.sql.ClientE2ETestSuite.afterAll(ClientE2ETestSuite.scala:23)
[info]   at org.scalatest.BeforeAndAfterAll.$anonfun$run$1(BeforeAndAfterAll.scala:225)
[info]   at org.scalatest.Status.$anonfun$withAfterEffect$1(Status.scala:377)
[info]   at org.scalatest.Status.$anonfun$withAfterEffect$1$adapted(Status.scala:373)
[info]   at org.scalatest.CompositeStatus.whenCompleted(Status.scala:962)
[info]   at org.scalatest.Status.withAfterEffect(Status.scala:373)
[info]   at org.scalatest.Status.withAfterEffect$(Status.scala:371)
[info]   at org.scalatest.CompositeStatus.withAfterEffect(Status.scala:863)
[info]   at org.scalatest.BeforeAndAfterAll.run(BeforeAndAfterAll.scala:224)
[info]   at org.scalatest.BeforeAndAfterAll.run$(BeforeAndAfterAll.scala:208)
[info]   at org.apache.spark.sql.ClientE2ETestSuite.run(ClientE2ETestSuite.scala:23)
[info]   at org.scalatest.tools.Framework.org$scalatest$tools$Framework$$runSuite(Framework.scala:321)
[info]   at org.scalatest.tools.Framework$ScalaTestTask.execute(Framework.scala:517)
[info]   at sbt.ForkMain$Run.lambda$runTest$1(ForkMain.java:413)
[info]   at java.util.concurrent.FutureTask.run(FutureTask.java:266)
[info]   at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
[info]   at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
[info]   at java.lang.Thread.run(Thread.java:750)
[info] Run completed in 9 seconds, 695 milliseconds.
[info] Total number of tests run: 4
[info] Suites: completed 0, aborted 1
[info] Tests: succeeded 4, failed 0, canceled 0, ignored 0, pending 0
[info] *** 1 SUITE ABORTED ***
[error] Error during tests:
[error] 	org.apache.spark.sql.ClientE2ETestSuite

```

Reply:
What is GA test BTW?

Reply:
GitHub Action ...